### PR TITLE
Add no-cache option

### PR DESCRIPTION
**Pull request by maschwenk**
When using Self Hosted Runners, bandwidth is not a huge concern, and the time it takes to compute/store/restore a Github Actions cache (which costs money) is pretty negligible for a binary for ~30 MiB.

There's a lot of restrictions to the Github Cache that make it kinda not super helpful in a lot of contexts as well:

- https://github.com/actions/setup-node/issues/663

https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache

The moment I started using this Action, I got a flood of cache entries:

image

Even though the cache key (the url) is completely stable

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@ inputs:
   scope:
     required: false
     description: "The scope for authenticating with the package registry."
+  disable-cache:
```
Comment (Electroid) [inline action.yml]: I think it makes sense to call this `no-cache`

**Comment by maschwenk** [comment]
@Electroid addressed those changes, also integrated some of the changes from:

- https://github.com/oven-sh/setup-bun/pull/62

by @Akalanka47000

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "private": true,
   "name": "setup-bun",
-  "version": "1.1.1",
+  "version": "1.2.0",
```
Comment (maschwenk) [inline package.json:4]: Unsure if this is better left to you guys

**Review by Electroid — APPROVED** [review]
Thanks for working on this, looks good!

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
CI failing

**Comment by maschwenk** [comment]
I think I fixed it, but let me also take a look at running these same CI tests on my branch or by just triggering from another repo. Apologies.

**Comment by maschwenk** [comment]
@Electroid @Jarred-Sumner I think we're good now!

**Review by Electroid — APPROVED** [review]
CI now passes, thanks!

**Pull request merged**